src/JSONLD/Processor.ts

```diff
@@ -0,0 +1,435 @@
+import Error from "../Errors/InvalidJSONLDSyntaxError";
+import * as Errors from "./../Errors";
+import * as HTTP from "./../HTTP";
+import * as ObjectSchema from "./../ObjectSchema";
+import * as RDF from "./../RDF";
+import * as Utils from "./../Utils";
+
+const MAX_CTX_URLS:number = 10;
+const LINK_HEADER_REL:string = "http://www.w3.org/ns/json-ld#context";
+
+export class Class {
+	static expand( input:Object ):Promise<Array<Object>> {
+		// Find and resolve context URLs
+		return _retrieveContexts( input, <{[ index:string ]:boolean}> Object.create( null ), "" ).then( () => {
+			// Expand the document
+			let expanded:any = _process( new ObjectSchema.DigestedObjectSchema(), input );
+
+			// Optimize @graph
+			if( Utils.isObject( expanded ) && "@graph" in expanded && Object.keys( expanded ).length === 1 ) {
+				expanded = expanded[ "@graph" ];
+			} else if ( expanded === null ) {
+				expanded = [];
+			}
+
+			// Normalize to an array
+			if( ! Utils.isArray( expanded ) ) expanded = [ expanded ];
+
+			return expanded;
+		} );
+	}
+}
+
+function _getTargetFromLinkHeader( header:HTTP.Header.Class ):string {
+	let rLinkHeader:RegExp = /\s*<([^>]*?)>\s*(?:;\s*(.*))?/;
+	for( let value of header.values ) {
+		let match:string[] = value.toString().match( rLinkHeader );
+		if( ! match ) continue;
+
+		let target:string = match[ 1 ];
+		let params:string = match[ 2 ];
+		let rParams:RegExp = /(.*?)=(?:(?:"([^"]*?)")|([^"]*?))\s*(?:(?:;\s*)|$)/g;
+
+		let result:{[ key:string ]:string} = {};
+		while( true ) {
+			match = rParams.exec( params );
+			if( ! match ) break;
+
+			result[ match[ 1 ] ] = ( match[ 2 ] === undefined ) ? match[ 3 ] : match[ 2 ];
+		}
+		if( result[ "rel" ] === LINK_HEADER_REL ) return target;
+	}
+	return null;
+}
+
+function _findContextURLs( input:Object, contexts:{[ index:string ]:Object}, base:string, replace:boolean = false ):boolean {
```

Comment:
Please remove the `_` from the function names and move them inside of the Processor class as `private static` methods